### PR TITLE
Update BDK version to 0.12.0

### PR DESCRIPTION
**Pull request by rajarshimaitra**
### Description

This PR updates the upstream BDK version to latest release (`v0.10.0`). 

### Notes to the reviewers

The latest release of BDK includes some breaking changes, some of which are handled here. A previously available function in BDK, `maintain_single_recipient()` is changed into `allow_shrinking()` which requires a scriptpubkey to be specified. So far I couldn't find any public API in BDK using which a scriptpubkey (or the transaction) can be retrieved from the wallet given a TxID. Marked the code with a `TODO`. It probably needs some additional API in BDK. Made an issue (https://github.com/bitcoindevkit/bdk/issues/427) to track this.

BDK `esplora` blockchain module has been divided into two types `using-ureq` and `using-reqwest`. `reqwest` is an async feature in BDK, which produces `Future` related errors in `bdk-cli` (https://github.com/bitcoindevkit/bdk-cli/pull/36#issue-702356592). Trying to solve that error creates more complexity (async related weridness). Thus decided to scrap `reqwest` and go with blocking `ureq` version, which I think is adequate for the usecases of `bdk-cli`.

Updated electrum cli args.     

#### All Submissions:

* [x] I've signed all my commits
* [x] I followed the [contribution guidelines](https://github.com/bitcoindevkit/bdk-cli/blob/master/CONTRIBUTING.md)
* [x] I ran `cargo fmt` and `cargo clippy` before committing

#### New Features:

* [ ] I've added tests for the new feature
* [ ] I've added docs for the new feature
* [ ] I've updated `CHANGELOG.md`

#### Bugfixes:

* [ ] This pull request breaks the existing API
* [ ] I've added tests to reproduce the issue which are now passing
* [ ] I'm linking the issue being fixed by this PR

### CODE REVIEW

File: Cargo.toml
Comment (rajarshimaitra) [reply Cargo.toml:15]: Good Idea. Better to enable `tls` in `ureq` if we use it. It seems like a general requirement to be able to connect https. Will open a PR if we agree..

Comment (notmandatory) [reply Cargo.toml:15]: True, maybe best of have `tls` feature enabled by default in `bdk`, but first I'd like to get @tcharding's input since he originally added the `ureq` dependency and may have an argument for leaving `tls` off by default.

Comment (tcharding) [reply Cargo.toml:15]: I have no problem with enabling TLS by default. As for why I disabled it, I can't remember exactly why, I probably just disabled default features during a debugging session and never turned TLS back on. Thanks for checking in with me and thanks for giving it the thought to realize we should enable it by default!

File: Cargo.toml
Comment (afilini) [inline Cargo.toml:35]: I'm trying to use this to test bitcoindevkit/bdk#429: to get bdk-cli to work in the browser we need a way to enable `bdk/use-esplora-reqwest`. I would suggest maybe adding another feature `esplora-reqwest` that we can use from wasm32

Comment (afilini) [reply Cargo.toml:35]: Or maybe even better: `async-esplora`

Comment (notmandatory) [reply Cargo.toml:35]: I think this PR may be blocked until @tcharding and @LLFourn  work out things out with https://github.com/bitcoindevkit/bdk/pull/433.

Comment (rajarshimaitra) [reply Cargo.toml:35]: Yes. I am not really sure how big the httpclient fix would be. If it takes much longer,  can we consider merging https://github.com/bitcoindevkit/bdk/pull/430 in the mean time? #36  is also pending on this one.

And I think that will fix @afilini 's issue too.

Comment (afilini) [reply Cargo.toml:35]: In the meantime I made this branch just to test the playground: https://github.com/afilini/bdk-cli/tree/update-with-reqwest

**Comment by rajarshimaitra** [comment]
Added esplora-reqwest into bdk-cli. 

@notmandatory we have to use `bdk-master` for now, until the esplora-fix is released upstream. This can be changed in next bdk release. 

@afilini please check if this meets your requirement, or something more to be done?

**Comment by notmandatory** [comment]
When `bdk` release `0.12.0` is out ~Sept-29 I think this one will be ready to merge. I've updated the title. :wink:

**Comment by rajarshimaitra** [comment]
@notmandatory  just FYI, the rest of the open PRs builds on top of this. So this needs to go first before they can be merged.. :)

**Comment by notmandatory** [comment]
I just made the [`bdk 0.12.0` release](https://crates.io/crates/bdk), so you're good to go on this one.

**Comment by rajarshimaitra** [comment]
Rebased with bdk v0.12. 

FYI, I am not updating #42 and #36 right now. For simplicity of merging this needs to be merged first. Then the rest two can be rebased on top of updated master. I think that will simplify the review process as all these PRs are interdependent.

**Review by notmandatory — APPROVED** [review]
re-ACK f9a96062bce59b93e3449bc812c1a9d5d2ad684a

I manually tested and works fine as long as correct esplora server URL is used. Either README needs to be updated, defaults need to be changed for `testnet` or maybe something else. Will merge this now so other PRs can proceed and we can fix the esplora server URL issue in a subsequent PR.

**Pull request merged**
